{{cookiecutter.github_project_name}}/setup.py

```diff
@@ -1,17 +1,153 @@
+
+def update_package_data(distribution):
+    """update package_data to catch changes during setup"""
+    build_py = distribution.get_command_obj('build_py')
+    # distribution.package_data = find_package_data()
+    # re-init build_py options which load package_data
+    build_py.finalize_options()
+
+
+class NPM(Command):
+    description = 'install package.json dependencies using npm'
+
+    user_options = []
+
+    node_modules = os.path.join(node_root, 'node_modules')
+
+    targets = [f for f in static_files if f.endswith('.js')]
+
+    def initialize_options(self):
+        pass
+
+    def finalize_options(self):
+        pass
+
+    def has_npm(self):
+        try:
+            check_call(['npm', '--version'])
+            return True
+        except:
+            return False
+
+    def should_run_npm_install(self):
+        package_json = os.path.join(node_root, 'package.json')
+        node_modules_exists = os.path.exists(self.node_modules)
+        return self.has_npm()
+
+    def run(self):
+        has_npm = self.has_npm()
+        if not has_npm:
+            log.error("`npm` unavailable.  If you're running this command using sudo, make sure `npm` is available to sudo")
+
+        env = os.environ.copy()
+        env['PATH'] = npm_path
+
+        if self.should_run_npm_install():
+            log.info("Installing build dependencies with npm.  This may take a while...")
+            check_call(['npm', 'install'], cwd=node_root, stdout=sys.stdout, stderr=sys.stderr)
+            os.utime(self.node_modules, None)
+
+        for t in self.targets:
+            if not os.path.exists(t):
+                msg = 'Missing file: %s' % t
+                if not has_npm:
+                    msg += ('\nnpm is required to build a development version '
+                            '{{ cookiecutter.python_package_name }}')
+                raise ValueError(msg)
+
+        # update package data in case this created new files
+        update_package_data(self.distribution)
+
+setup_args = {
+    'name': '{{ cookiecutter.python_package_name  }}',
+    'description': '{{ cookiecutter.project_short_description }}',
+    'long_description': LONG_DESCRIPTION,
+    'include_package_data': True,
+    'data_files': [
+        ('share/jupyter/labextensions/{{ cookiecutter.npm_package_name }}',
+         static_files),
+    ],
+    'packages': find_packages(),
+    'zip_safe': False,
+    'cmdclass': {
+        'build_py': js_prerelease(build_py),
+        'egg_info': js_prerelease(egg_info),
+        'sdist': js_prerelease(sdist, strict=True),
+        'jsdeps': NPM,
+    },
+
+    'author': '{{ cookiecutter.author_name }}',
+    'author_email': '{{ cookiecutter.author_email }}',
+    'url': 'https://github.com/{{ cookiecutter.github_organization_name }}/{{ cookiecutter.github_project_name }}',
+    'keywords': [
+        'jupyterlab',
+        'jupyter',
+    ],
+}
+
+# Look for a versioneer-style _version.py, not included in this cookiecutter.
```

Comment:
When you need to modify the (even super-thin) wrapper, e.g. because you rerender a cookiecutter template, you may want to increase the version number for the wrapper.